index.js

```diff
@@ -1,47 +1,50 @@
-addEventListener('fetch', event => {
-  event.respondWith(handle(event.request))
-})
-
-async function handle(request) {
-  if (request.method === "OPTIONS") {
-    return handleOptions(request)
-  } else if (request.method === "GET" ||
-             request.method === "HEAD" ||
-             request.method === "POST") {
-    // Pass-through to origin.
-    return fetch(request)
-  } else {
-    return new Response(null, {
-      status: 405,
-      statusText: "Method Not Allowed",
-    })
-  }
-}
-
-// We support the GET, POST, HEAD, and OPTIONS methods from any origin,
-// and accept the Content-Type header on requests. These headers must be
-// present on all responses to all CORS requests. In practice, this means
-// all responses to OPTIONS requests.
+/**
+ * This snippet supports requests with GET, POST, HEAD, or OPTIONS methods from any origin.
+ * These requests are able to view the Content-Type header only.
+ * Change the corHeaders below to modify the type of requests and headers
+ * to accept
+ */
+const allowedMethods = 'GET, HEAD, POST, OPTIONS'
 const corsHeaders = {
-  "Access-Control-Allow-Origin": "*",
-  "Access-Control-Allow-Methods": "GET, HEAD, POST, OPTIONS",
-  "Access-Control-Allow-Headers": "Content-Type",
+  'Access-Control-Allow-Origin': '*',
+  'Access-Control-Allow-Methods': allowedMethods,
+  'Access-Control-Allow-Headers': 'Content-Type',
 }
 
 function handleOptions(request) {
-  if (request.headers.get("Origin") !== null &&
-      request.headers.get("Access-Control-Request-Method") !== null &&
-      request.headers.get("Access-Control-Request-Headers") !== null) {
+  if (
+    request.headers.get('Origin') !== null &&
+    request.headers.get('Access-Control-Request-Method') !== null &&
```

Comment:
I believe we should check that the contents of `Access-Control-Request-Method` is in `allowedMethods` @ConzorKingKong

Reply:
```suggestion
    request.headers.get('Access-Control-Request-Method') !== null &&
    allowedMethods.contains(request.headers.get('Access-Control-Request-Method')) &&
```

Reply:
That's definitely a smart idea